src/taskEvents/script.js

```diff
@@ -0,0 +1,291 @@
+const overlay = document.querySelector('.overlay');
+
+const closeBtn = document.getElementById('close-btn');
+closeBtn.addEventListener('click', closeModal);
+const genericModalCloseBtn = document.getElementById('close-generic-modal');
+genericModalCloseBtn.addEventListener('click', closeGenericModal);
+
+async function createProfileModal(username) {
+  try {
+    overlay.style.display = 'block';
+    const { user } = await getUserData(username);
+    // another call for roles will be made when we have userSkills collection
+    const userImg = createElement({
+      type: 'img',
+      attributes: {
+        src: user.picture.url || './assets/Avatar.png',
+        alt: 'user img',
+        class: 'user-img',
+      },
+    });
+    const userName = createElement({
+      type: 'p',
+      attributes: { class: 'username' },
+      innerText: user.username,
+    });
+
+    document.querySelector('.top-div').prepend(userName);
+    document.querySelector('.top-div').prepend(userImg);
+
+    const skillTitle = createElement({
+      type: 'p',
+      attributes: { class: 'skill-title' },
+      innerText: 'Skills',
+    });
+    modal.appendChild(skillTitle);
+
+    const rolesArray = [
+      'React-level1',
+      'Ember-level2',
+      'Remix-level3',
+      'NodeJs-level3',
+      'random-levl1',
+    ];
+    const roles = createRolesDiv(rolesArray);
+    modal.appendChild(roles);
+
+    const activityBtn = createUserActivityBtn(username);
+    modal.appendChild(activityBtn);
+  } catch (err) {
+    addErrorMessage(modal);
+    console.log(err);
+  }
+}
+
+function createRolesDiv(roles) {
+  const Allroles = [...roles];
+  const rolesDiv = createElement({
+    type: 'div',
+    attributes: { class: 'roles-div' },
+  });
+  Allroles.map((role, index) => {
+    const element = createElement({
+      type: 'div',
+      attributes: { class: 'roles-div-item' },
+      innerText: role,
+    });
+    const removeBtn = createElement({
+      type: 'button',
+      attributes: { class: 'remove-btn', id: index },
+      innerText: 'x',
+    });
+    removeBtn.addEventListener('click', removeSkill);
+    element.appendChild(removeBtn);
+    rolesDiv.append(element);
+  });
+  const addBtn = createElement({
+    type: 'button',
+    attributes: { class: 'add-btn' },
+    innerText: '+',
+  });
+  addBtn.addEventListener('click', openAddSkillModal);
+  rolesDiv.appendChild(addBtn);
+  return rolesDiv;
+}
+
+function createUserActivityBtn(username) {
+  const activityBtnDiv = createElement({
+    type: 'div',
+    attributes: { class: 'activity-btn-div' },
+  });
+  const activityBtn = createElement({
+    type: 'button',
+    attributes: { class: 'activity-btn' },
+    innerText: 'show user activity',
+  });
+  activityBtn.addEventListener('click', () => openUserActivityModal(username));
+  activityBtnDiv.appendChild(activityBtn);
+  return activityBtnDiv;
+}
+
+function closeModal() {
+  overlay.style.display = 'none';
```

Comment:
can we add a CSS class to the element and then toggle between that class to hide or show the element?

Reply:
yeah make sense I will do that